pkgs/games/xash3d/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, lib
+, makeWrapper
+, cmake
+, pkg-config
+, SDL2
+, fontconfig
+, fetchgit
+, python3
+}:
+{ valve ? throw ''type valve directory for example nix-env --argstr valve "/home/max/valve" -iA xash3d'' } @ args:
```

Comment:
Why do we need compile this in?

Reply:
https://github.com/FWGS/xash3d/wiki/Building-and-running
valve folder should be in the same directory where is xash3d located. So the only way is to create symlink to this directory. This argument needs for valve directory path

Reply:
> valve folder should be in the same directory where is xash3d located. 

You can just set the XASH3D_BASEDIR variable. To make execution easier you can make a wrapper script that will load this variable from a config file and then run xash3d